operator/src/main/java/oracle/kubernetes/weblogic/domain/model/ServerPod.java

```diff
@@ -307,12 +311,18 @@ private void copyValues(V1Affinity to, V1Affinity from) {
   }
 
   private void copyValues(V1NodeAffinity to, V1NodeAffinity from) {
-    if (to.getPreferredDuringSchedulingIgnoredDuringExecution() == null) {
-      to.setPreferredDuringSchedulingIgnoredDuringExecution(from.getPreferredDuringSchedulingIgnoredDuringExecution());
-    } else if (from.getPreferredDuringSchedulingIgnoredDuringExecution() != null) {
-      from.getPreferredDuringSchedulingIgnoredDuringExecution()
-          .forEach(to::addPreferredDuringSchedulingIgnoredDuringExecutionItem);
+    if (from.getPreferredDuringSchedulingIgnoredDuringExecution() != null) {
+      List<V1PreferredSchedulingTerm> preferredDuringSchedulingIgnoredDuringExecution = new ArrayList<>();
+      if (to.getPreferredDuringSchedulingIgnoredDuringExecution() != null) {
+        preferredDuringSchedulingIgnoredDuringExecution =
+            Stream.concat(to.getPreferredDuringSchedulingIgnoredDuringExecution().stream(),
+                from.getPreferredDuringSchedulingIgnoredDuringExecution().stream())
+                .distinct()
+                .collect(Collectors.toList());
+      }
+      to.setPreferredDuringSchedulingIgnoredDuringExecution(preferredDuringSchedulingIgnoredDuringExecution);
```

Comment:
If `to.getPreferred..()` was `null`, does the array `preferredDuringSchedulingIgnoredDuringExecution` contains element from `from.getPreferred...()`?

Reply:
Good catch... This bug is also in the code I copied this from (look at copyValues(V1Capabilities, V1Capabilities)).

Reply:
@alai8, this is fixed... can you please recheck?